app/src/org/commcare/preferences/HiddenPreferences.java

```diff
@@ -637,4 +637,17 @@ public static void setInterruptedFormIndex(String serializedFormIndex) {
                 .putString(INTERRUPTED_FORM_INDEX + currentUserId,  serializedFormIndex)
                 .apply();
     }
+
+    public static String getInterruptedFormIndex() {
+        String currentUserId = CommCareApplication.instance().getCurrentUserId();
```

Comment:
Nit, but if available this should probably also be keyed to the App ID, since a user's ID is shared between two apps in the same project space.

Reply:
About this, is it still necessary to have the App ID if we are validating the FormIndex based on the SSD Id? Is there any other reason to still have it there?

Reply:
I think this should still be keyed on the app ID. The SSD Id is likely to have collisions on its own, since it's just a DB row id, right?

Reply:
I would think we should use form instance id , app id doesn't make a difference here as the preferences we are writing to are app specific.

Reply:
>  app id doesn't make a difference here as the preferences we are writing to are app specific.

Good point, taking that into considering plus that we only load the interrupted form index when recovering an expired session, which happens right after the next login, our current check seems to be sufficient. I have tested it with a few scenarios, such as logging back in an app from the same domain and with the same user, with form linking, and worked fine. Let me know if there are other scenarios we should be concerned about.  

As per the form instance Id, I think this it the `form record id` (right, @shubham1g5 ?), we can always add it to strengthen the validation, this is very low loe as the `SessionStateDescriptor` also stores the form record Id.

Reply:
>  we can always add it to strengthen the validation, this is very low loe as the SessionStateDescriptor also stores the form record Id.

@avazirna think the idea here is to make this collision resistant. Even though `SessionStateDescriptor` stores form record Id, here we are only storing the database id of the DB row for the corresponding  `SessionStateDescriptor`. This will be an Integer number and is not very collision resistive in case something goes wrong that lets say clear the `SessionStateDescriptor` table and cause it to reuse the integer ids. In contrast form instance ids are unique uuids which are much less likely to collide in context of a single app.